proxy/proxy.go

```diff
@@ -375,21 +376,23 @@ func (p *proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	start = time.Now()
-	response := c.Response()
-	if p.preserveOriginal {
-		c.originalResponse = cloneResponseMetadata(response)
+	if !c.served && p.preserveOriginal {
```

Comment:
this takes into account, that filters on the response chain can also call Serve() and override the response from the backend or from the previous 'serving' filter